lib/teleterm/daemon/config.go

```diff
@@ -25,10 +27,28 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+type (
+	cluster interface {
+		GetServers(ctx context.Context) ([]clusters.Server, error)
+	}
+	storage interface {
+		GetByURI(uri string) (cluster, error)
+	}
+)
+
+// This is used to wrap a real clusters.Storage so it meets our interface
+type StorageWrapper struct {
+	*clusters.Storage
+}
+
+func (sw *StorageWrapper) GetByURI(uri string) (cluster, error) {
+	return sw.GetByURI(uri)
```

Comment:
Looks like an infinite recursion to me?

```suggestion
	return sw.Storage.GetByURI(uri)
```